src/core/solvers/box-ddp.cpp

```diff
@@ -1,14 +1,19 @@
 ///////////////////////////////////////////////////////////////////////////////
 // BSD 3-Clause License
 //
-// Copyright (C) 2019-2020, CNRS-LAAS, University of Edinburgh
+// Copyright (C) 2019-2021, CNRS-LAAS, University of Edinburgh
 // Copyright note valid unless otherwise stated in individual files.
 // All rights reserved.
 ///////////////////////////////////////////////////////////////////////////////
 
 #include <iostream>
-#include "crocoddyl/core/utils/exception.hpp"
+#ifdef CROCODDYL_WITH_MULTITHREADING
+#include <omp.h>
+#define NUM_THREADS CROCODDYL_WITH_NTHREADS
```

Comment:
I think NUM_THREADS is too generic and can conflict with other libs. Always use CROCODDYL prefix

Reply:
Do we need to redefine `NUM_THREADS` here? We can directly use `CROCODDYL_WITH_NTHREADS` and skip re-definition of the macro.

Reply:
Yes, indeed, let's use `CROCODDYL_WITH_NTHREADS`. However, I was thinking that it might be useful to create a mechanism were users can change the number of threads per object, e.g.,
```python
problem = crocoddyl.ShootingProblem(x0, runningModels, terminalModel)
problem.set_nthreads(10) # otherwise use CROCODDYL_WITH_NTHREADS
solver = crocoddyl.SolverDDP(problem)
solver.set_nthreads(10) # same here
```

I am keen to implement this approach in this PR. What do you think?

Reply:
I had this in preparation as a PR to set/get the threads for shooting problem (ifdef'd wrapper around `omp_set_num_threads`) - I'll let you handle it then or happy to add to this PR.

Reply:
Removed `NUM_THREADS` as requested by @jcarpent and @proyan 

> I had this in preparation as a PR to set/get the threads for shooting problem (ifdef'd wrapper around `omp_set_num_threads`) - I'll let you handle it then or happy to add to this PR.

Yes, let me handle it. I had started to work on it. Soon I will push the code :)
Thanks in any case!

Reply:
I have pushed the code that allows user to modify the nthreads.
I will close this issue as I believe it is resolved :)